src/Kernels/DeviceAux/cuda/PlasticityAux.cu

```diff
@@ -0,0 +1,235 @@
+#include <Kernels/DeviceAux/PlasticityAux.h>
+#include <init.h>
+#include <cmath>
+
+#if REAL_SIZE == 8
```

Comment:
Is this the way to do it? Disclaimer: I'm no expert on Cuda.

Reply:
Hi, Could you tell me which part of the code you refer to?

Do you mean this one?
https://github.com/SeisSol/SeisSol/blob/27140584ba70cf88c46f26f348414117a82938b8/src/Kernels/DeviceAux/cuda/PlasticityAux.cu#L5-L11

Reply:
Yes these defines. Is this a standard way of writing stuff in cuda?

Reply:
So, I agree that it is ugly but it is necessary in order to support both double and single precision.
Here is a definition of `max` function in cuda
`__device__ ​ double fmax ( double , double ) `  [reference double precision](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__DOUBLE.html#group__CUDA__MATH__DOUBLE)
`__device__ ​ float fmaxf ( float  x, float  y )`  [reference single precision](https://docs.nvidia.com/cuda/cuda-math-api/group__CUDA__MATH__SINGLE.html#group__CUDA__MATH__SINGLE)

I don't know why CUDA devs didn't use templates in this case. Officially, CUDA supports templates but maybe math functions are implemented as intrinsics